### PR TITLE
Cortical Borers [Needs so much rebalancing it isn't funny]

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Readds cortical borers, which were removed by /tg/ some time ago. These are updated/modernized to the best of my ability, but the balance is extremely out-of-whack and needs to be adjusted. Suggestions welcome

## Why It's Good For The Game
More antags and things ooga

image
*tasty*

## Changelog
:cl:
add: Cortical borers, commonly known as brain slugs, have appeared in NT space! Be cautious, as they could be hiding in any of your fellow employees!
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -153,6 +153,12 @@ GENE SCANNER
 		if(H.undergoing_cardiac_arrest() && H.stat != DEAD)
 			render_list += "<span class='alert'>Subject suffering from heart attack: Apply defibrillation or other electric shock immediately!</span>\n"
 
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
+		if(C.has_brain_worms())
+			render_list += "<span class='danger'>Foreign organism detected in subject's cranium. Recommended treatment: Dosage of sucrose solution and removal of object via surgery.</span>\n"
```
Comment (Superyodeler) [inline code/game/objects/items/devices/scanners.dm]:
```suggestion
	if(iscarbon(M))		//Wasp Begin - Borers
		var/mob/living/carbon/C = M
		if(C.has_brain_worms())
			render_list += "<span class='danger'>Foreign organism detected in subject's cranium. Recommended treatment: Dosage of sucrose solution and removal of object via surgery.</span>\n"		//Wasp End
```

File: code/modules/antagonists/changeling/powers/panacea.dm [hunk 2]
```diff
@@ -26,6 +26,16 @@
...
 
+	var/mob/living/simple_animal/borer/B = user.has_brain_worms()
+	if(B)
+		if(B.controlling)
+			B.detatch()
+		B.leave_victim()
+		if(iscarbon(user))
+			var/mob/living/carbon/C = user
+			C.vomit(0, toxic = TRUE)
+			to_chat(user, "<span class='notice'>A parasite exits our form.</span>")
```
Comment (Superyodeler) [inline code/modules/antagonists/changeling/powers/panacea.dm:37]:
```suggestion
	var/mob/living/simple_animal/borer/B = user.has_brain_worms()		//Wasp Begin - Borers
	if(B)
		if(B.controlling)
			B.detatch()
		B.leave_victim()
		if(iscarbon(user))
			var/mob/living/carbon/C = user
			C.vomit(0, toxic = TRUE)
			to_chat(user, "<span class='notice'>A parasite exits our form.</span>")		//Wasp End
```

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 3]
```diff
@@ -327,7 +327,10 @@
 	return list()
 
 /obj/machinery/atmospherics/update_remote_sight(mob/user)
-	user.sight |= (SEE_TURFS|BLIND)
+	if(isborer(user))
+		user.sight |= (SEE_PIXELS)
+	else
+		user.sight |= (SEE_TURFS|BLIND)
```
Comment (Superyodeler) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:333]:
```suggestion
	if(isborer(user))		//Wasp Begin - Borers
		user.sight |= (SEE_PIXELS)
	else
		user.sight |= (SEE_TURFS|BLIND)		//Wasp End

File: code/modules/client/verbs/suicide.dm [hunk 4]
```diff
@@ -258,4 +258,7 @@
 	if(!(mobility_flags & MOBILITY_USE))	//just while I finish up the new 'fun' suiciding verb. This is to prevent metagaming via suicide
 		to_chat(src, "<span class='warning'>You can't commit suicide whilst immobile! ((You can type Ghost instead however.))</span>")
 		return
+	if(has_brain_worms())
+		to_chat(src, "You can't bring yourself to commit suicide!")
+		return
```
Comment (Superyodeler) [inline code/modules/client/verbs/suicide.dm:263]:
```suggestion
	if(has_brain_worms())		//Wasp Begin - Borers
		to_chat(src, "You can't bring yourself to commit suicide!")
		return		//Wasp End
```

File: code/modules/mob/living/brain/brain_item.dm [hunk 5]
```diff
@@ -58,6 +58,15 @@
...
 /obj/item/organ/brain/Remove(mob/living/carbon/C, special = 0, no_id_transfer = FALSE)
+	if(!special)
+		if(C.has_brain_worms())
+			var/mob/living/simple_animal/borer/B = C.has_brain_worms()
+			if(B.controlling)
+				B.victim.release_control()
+				to_chat(B, "<span class='userdanger'>Your probiscis is ripped out as your host's brain is removed!</span>")
+				B.apply_damage(15)
+			B.leave_victim()
+
```
Comment (Superyodeler) [inline code/modules/mob/living/brain/brain_item.dm:69]:
```suggestion
	if(!special)		//Wasp Begin - Borers
		if(C.has_brain_worms())
			var/mob/living/simple_animal/borer/B = C.has_brain_worms()
			if(B.controlling)
				B.victim.release_control()
				to_chat(B, "<span class='userdanger'>Your probiscis is ripped out as your host's brain is removed!</span>")
				B.apply_damage(15)
			B.leave_victim()		//Wasp End
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 6]
```diff
@@ -64,6 +64,10 @@
 				stat(null, "Tank Pressure: [internal.air_contents.return_pressure()]")
 				stat(null, "Distribution Pressure: [internal.distribute_pressure]")
 
+		var/mob/living/simple_animal/borer/B = has_brain_worms()
+		if(B && B.controlling)
+			stat("Chemicals", B.chemicals)
```
Comment (Superyodeler) [inline code/modules/mob/living/carbon/human/human.dm:69]:
```suggestion
		var/mob/living/simple_animal/borer/B = has_brain_worms()		//Wasp Begin - Borers
		if(B && B.controlling)
			stat("Chemicals", B.chemicals)		//Wasp End
```

File: code/modules/surgery/organ_manipulation.dm [hunk 7]
```diff
@@ -137,6 +137,13 @@
 	else if(implement_type in implements_extract)
 		current_type = "extract"
 		var/list/organs = target.getorganszone(target_zone)
+
+		var/mob/living/simple_animal/borer/B = target.has_brain_worms()
+		if(target.has_brain_worms())
+			user.visible_message("[user] begins to extract [B] from [target]'s [parse_zone(target_zone)].",
+					"<span class='notice'>You begin to extract [B] from [target]'s [parse_zone(target_zone)]...</span>")
+			return TRUE
```
Comment (Superyodeler) [inline code/modules/surgery/organ_manipulation.dm:145]:
```suggestion
		var/mob/living/simple_animal/borer/B = target.has_brain_worms()		//Wasp Begin - Borers
		if(target.has_brain_worms())
			user.visible_message("[user] begins to extract [B] from [target]'s [parse_zone(target_zone)].",
					"<span class='notice'>You begin to extract [B] from [target]'s [parse_zone(target_zone)]...</span>")
			return TRUE		//Wasp End
```

File: waspstation/code/modules/antagonists/borer/borer.dm [hunk 8]
```diff
@@ -75,6 +75,7 @@ GLOBAL_VAR_INIT(total_borer_hosts_needed, 3)
 	attack_verb_simple = "chomp"
 	attack_sound = 'sound/weapons/bite.ogg'
 	pass_flags = PASSTABLE | PASSMOB | PASSDOORHATCH
+	layer = UNDERDOOR
```
Comment (Superyodeler) [inline waspstation/code/modules/antagonists/borer/borer.dm:78]: You fixed this while I was testing

Comment (MarkSuckerberg) [reply waspstation/code/modules/antagonists/borer/borer.dm:78]: yerp, I saw it and just fixed it not a few minutes ago

**Comment by TheNeoGamer42** [comment]
this bots lame why did it never get merged anyways, its been testmerged for awhile, or atleast since I last bothered to check the testmerges, just fullmerge

**Comment by Superyodeler** [comment]
Did the borer sprites get moved up so they don't peek out from under tables?

**Comment by MarkSuckerberg** [comment]
Fixing

**Comment by TheNeoGamer42** [comment]
merge when

**Comment by MarkSuckerberg** [comment]
Split personality bug probably fixed, but needs testing.

**Comment by SynnGraffkin** [comment]
Maybe have Borer eggs be a thing as well, and add it to the Xenospawn pool in Xenobio pen at a higher rate?

https://github.com/vgstation-coders/vgstation13/blob/9867343a916671f8a074d72f641821de5f92b22a/code/modules/mob/living/simple_animal/borer/egg.dm

**Comment by TheNeoGamer42** [comment]
what the fuck is this bot doing closing good PRs that everyone forgot about, *scream

**Comment by Superyodeler** [comment]
Bot pls

File: waspstation/code/modules/antagonists/borer/borer.dm [hunk 9]
```diff
@@ -0,0 +1,1254 @@
...
+
+	usr << browse(null, "window=ViewBorer\ref[src]Chems;size=600x800")
+	usr << browse(html, "window=ViewBorer\ref[src]Chems;size=600x800")
+
+	return
+
+/mob/living/simple_animal/borer/verb/hide()
+	set category = "Borer"
+	set name = "Hide"
+	set desc = "Become invisible to the common eye."
```
Comment (Superyodeler) [inline waspstation/code/modules/antagonists/borer/borer.dm:490]:
```suggestion
	set desc = "Allows borers to hide beneath tables or certain items. Toggled on or off."
```

File: waspstation/code/modules/antagonists/borer/borer.dm [hunk 10]
```diff
@@ -0,0 +1,1254 @@
...
+	desc = "Infest a suitable humanoid host."
+	button_icon_state = "infest"
+
+/datum/action/innate/borer/infest_host/Activate()
+	var/mob/living/simple_animal/borer/B = owner
+	B.infect_victim()
+
+/datum/action/innate/borer/toggle_hide
+	name = "Toggle Hide"
+	desc = "Become invisible to the common eye. Toggled on or off."
```
Comment (Superyodeler) [inline waspstation/code/modules/antagonists/borer/borer.dm:1021]:
```suggestion
	desc = "Allows borers to hide beneath tables or certain items. Toggled on or off."
```

**Comment by Superyodeler** [comment]
Borerer

**Comment by SynnGraffkin** [comment]
Marge it

**Pull request merged**